packages/app/src/utils/isValidHost.ts

```diff
@@ -26,10 +26,12 @@ export const isValidHost = ({
   forceProductionEnv?: boolean
   selfHostedSlug?: string | null
 }): boolean => {
-  const { slug, kind } = getInfoFromJwt(accessToken)
+  const decodedJWT = jwtDecode(accessToken)
+  const slug = jwtIsSalesChannel(decodedJWT.payload)
+    ? decodedJWT.payload.organization.slug
+    : ''
 
-  const isInvalidChannel = kind !== 'sales_channel'
-  if (isInvalidChannel) {
+  if (!jwtIsSalesChannel(decodedJWT.payload)) {
```

Comment:
You could move this if statement right before the `const slug = ...`, this way the slug will always be present.

Reply:
Sure! Changes applied!